tests/integration/helpers/stop-propagation-test.js

```diff
@@ -0,0 +1,115 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, click } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+
+module('Integration | Helper | stop-propagation', function(hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function() {
+    this.onAncestorClick = () => {
+      throw new Error(`Uncaught 'click' event.`);
+    };
+  });
+
+  // The rest of this test suite relies on this.
+  test('a click event propagates up the DOM by default', async function(assert) {
+    assert.expect(1);
+
+    this.onAncestorClick = () => {
+      assert.ok(true);
+    };
+
+    await render(hbs`
+      <div {{on "click" this.onAncestorClick}}>
+        <button>Click me</button>
+      </div>
+    `);
+
+    await click('button');
+  });
+
+  test('{{on "click" (stop-propagation)}}', async function(assert) {
+    assert.expect(0);
+
+    await render(hbs`
+      <div {{on "click" this.onAncestorClick}}>
+        <button {{on "click" (stop-propagation)}}>Click me</button>
+      </div>
+    `);
+
+    await click('button');
+  });
+
+  test('{{on "click" (stop-propagation) capture=true}}', async function(assert) {
+    assert.expect(1);
+
+    const calls = [];
```

Comment:
Totally not an issue, but I wanted to mention that I've started using `assert.step` / `assert.verifySteps` for these sorts of "order tracking" use cases. This test would look like:

```js
  test('{{on "click" (stop-propagation) capture=true}}', async function(assert) {
    assert.expect(1);

    this.outerListener = () => assert.step('outer');
    this.innerListener = () => assert.step('inner');

     await render(hbs`
      <div {{on "click" (stop-propagation this.outerListener) capture=true}}>
        <button {{on "click" this.innerListener}}>inner</button>
      </div>
    `);

     await click('button');

     assert.verifySteps(['outer'], 'it only runs the outer listener');
  });
```

See:

* https://api.qunitjs.com/assert/step
* https://api.qunitjs.com/assert/verifySteps

Reply:
Didn’t know about this! Already have ideas how I can use it in other projects. Thanks!